src/http/formdata.cr

```diff
@@ -138,13 +138,15 @@ module HTTP::FormData
     size = nil
     name = nil
 
-    parts = content_disposition.split(';')
-    type = parts[0]
-    raise Error.new("Invalid Content-Disposition: not form-data") unless type == "form-data"
-    (1...parts.size).each do |i|
-      part = parts[i]
+    first = true
+    content_disposition.split(';') do |part|
+      if first
+        first = false
+        raise Error.new("Invalid Content-Disposition: not form-data") unless part == "form-data"
+        next
+      end
```

Comment:
**suggestion:**
This is much cleaner and performance characteristics should be comparable:
```suggestion
    parts = content_disposition.lchop?("form-data;")
    raise Error.new("Invalid Content-Disposition: not form-data") unless parts
    parts.split(';') do |part|
```

Reply:
That would create a near copy of `content_disposition` which may consume even more memory than the original implementation.

Maybe the existing use of `.split` isn't so bad after all?

Reply:
Good point. Maybe we could avoid that allocation by using `starts_with?` and nothing else. Unknown keys are implicitly skipped in the loop.
However, there is the issue of potential white space around the separators (not sure if that's valid though).
So yeah probably we should better leave this unchanged for now. It's growing too much complexity for a simple generic refactor.